### ISSUE TITLE
Attempting to make water follow player

### ISSUE THREAD

**Issue by coopersully** [Guayabazo/DynamicWater#1]
**Explanation:**
The buoyancy system and boat controller both work perfectly fine across the world, even when the mesh isn't around the player, leading me to believe that if I can just somehow modify the scripts to only render the waves that are *near* the player, perhaps using the **dimension** value as a radius, I could make a seemingly "infinite ocean" of the water system you've made. The issue I'm trying to solve is: how do I get the waves themselves to follow the player, so that the set **dimension value** is the radius of the waves around the player?

**TL;DR** I'm trying to use this to create a 'infinite water' world where the water is always below the player, but I'm not quite sure how to do it.

**Comment by Guayabazo** [comment]
Hi Cooper!
First of all I am going to tell you that I am a self learner and maybe my
scripts are not the best. But I am doing my best!

I have been trying myself to do something like you are saying. Infinite
water mesh. But I have no idea how to make it if I want this kind of
ripples on the surface. I was thinking about making something like 9 meshes
(one in the middle, where the player is, and 8 around it) and change their
position as needed. BUT. I am approaching this method as an elastic mesh,
like a trampoline and I am afraid that meshes are not tileable this way.
Besides, it will need something to tell the ripples to continue in the next
mesh (tile). I do not think it is going to be possible.

*How do I get the waves themselves to follow the player, so that the
set dimension value is the radius of the waves around the player?*

I was thinking too about moving the mesh with the player, but ripples are
going to be very broken and they are the whole purpose of this experiment.
You would have to change the way the mesh is generated because it is
generated from 0 to 50 (on x and z). You could change the script to
generate the mesh from -25 to 25 (on x and z). This way the transform of
the mesh is in the center. After that you can make its position (on x and
z) the same as the player position.

*Or you can create a new Game Object and make it the water parent. And
change the water local position until the parent is centered. Then create a
script on the parent to move it with the player (on x and z).*

But I'll tell you. Moving around the water is going to look weird because
the waves and ripples are in local space (so to say) and they are going to
move with the water.

I think I am going to have to think of another method to make infinite
water with waves and ripples.
As you can see I am pretty much in the same spot as you are.

*I am going to share with you some tutorials that have been very useful for
me:*

https://www.youtube.com/watch?v=eL_zHQEju8s&t=3s&ab_channel=TomWeiland

https://www.youtube.com/watch?v=iasDPyC0QOg&ab_channel=ABeginner%27sDevVlog

https://catlikecoding.com/unity/tutorials/flow/waves/

https://www.youtube.com/watch?v=_Ij24zRI9J0&ab_channel=DitzelGames

*I will be very glad to hear from you if you have any ideas about how to
improve this water or a different approach to achieve a similar result.*
*Have a very nice week!*

P.D. If you are curious here you can play my first video game for Android : The
Last Flight
<https://play.google.com/store/apps/details?id=com.GuayabazoGames.TheLastFlight>

**Comment by Guayabazo** [comment]
Hi Cooper!
I have just uploaded a new water shader that does what we were looking for.
Here it is:

https://github.com/Guayabazo/InfiniteDynamicWaterURP

You can see the full video here:

https://www.youtube.com/watch?v=gZW1fGHpduA
